### ISSUE TITLE
Downloads and temp folder

### ISSUE THREAD

**Issue by alloween** [dnsev-h/eze#4]
I am assuming when eze downloads a gallery it first goes into the appdata>local>temp folder directory before the final archive is saved according to the user's preferences. This raises security/privacy concerns, as not all users would want evidence of their downloads floating around their temp folder on the system drive. I suggest adding the ability to modify this behavior and direct all downloaded data to a directory of the user's choosing.

**Comment by dnsev-h** [comment]
As far as I'm aware, that's not how it works. It should be stored in RAM for the most part. If the browser caches these ```Blob```/```ArrayBuffer``` objects as temp files somewhere, that's beyond my control. And even if it does do that, they shouldn't be immediately discernible as to their contents.

It should be noted that the final ```.zip``` file is not actually generated until every image is loaded into memory. This is the only file that I could imagine actually creating, and that file should be removed once the page is closed or reloaded.

In short, individual images should never appear as jpegs anywhere on your file system.

https://github.com/dnsev-h/eze/blob/master/src/eze.dev.user.js#L964
https://github.com/dnsev-h/eze/blob/master/src/eze.dev.user.js#L6383

**Comment by alloween** [comment]
Thank you very much for the clarification. I was unclear on the process, and assumed the data must be getting fully saved to the physical drive during download, before choosing a final save location. Sorry for taking up your time with a non-issue.

I assume what would happen, in the case of an exceptionally large gallery pushing a system's memory to the limit, is that the browser might then cache the data as temp files. But that data wouldn't be immediately discernible, as you say.

**Comment by dnsev-h** [comment]
In my experience, that probably doesn't happen either. The images are stored as raw bytes in RAM; the browser shouldn't be able to know this memory is for a jpeg or png or anything like that.

I've tested downloading that results in ```.zip``` file sizes in the 200 to 300 MiB range without issue; depends on how much memory your browser has available. The worst that seems to happen is the script will error when not enough RAM is available to create the ```.zip``` file.

That sort of error is hard to detect <i>if</i> it will happen before it actually happens, and at this time there is no recovery feature for this error. It can be avoided by downloading exceptionally large galleries in ranges of pages, or sometimes by simply restarting your browser.
